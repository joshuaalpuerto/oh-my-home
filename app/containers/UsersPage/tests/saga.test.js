
import { put } from 'redux-saga/effects'

import usersPageSagas, {
  fetchUsersSagas,
  fetchUsers
} from '../saga'
import {
  setUsersActions
} from '../actions'

const args = {}
const results = {
  id: 'supertestdev@gmail.com',
  email: 'supertestdev@gmail.com',
  name: {
    first: 'test',
    last: 'dev'
  }
}

describe('HomePage Saga', () => {
  describe('fetchUsersSagas', () => {
    let fetchUsersSagasMock

    beforeEach(() => {
      fetchUsersSagasMock = fetchUsersSagas()
      const callFetchLatest = fetchUsersSagasMock.next().value
      expect(callFetchLatest).toMatchSnapshot()
    })

    describe('fetchUsers', () => {
      // pass empty object for params.
      const fetchUsersMock = fetchUsers(args)

      it('should match the snapshot for setting token', () => {
        const settingToken = fetchUsersMock.next().value
        expect(settingToken).toMatchSnapshot()
      })

      it('should set setUsersAction', () => {
        // eslint-disable-next-line no-unused-vars
        const settingData = fetchUsersMock.next({ results: [results] }).value
        expect(settingData).toEqual(
          put(setUsersActions([results]))
        )
      })
    })
  })

  describe('defaultGenerators', () => {
    const usersGenerator = usersPageSagas()

    it('should have 1 functions', () => {
      const watchers = usersGenerator.next().value
      expect(watchers.length).toEqual(1)
    })
  })
})
