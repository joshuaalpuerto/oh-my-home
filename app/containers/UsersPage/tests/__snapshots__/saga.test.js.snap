// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Saga fetchUsersSagas fetchUsers should match the snapshot for setting token 1`] = `
Object {
  "@@redux-saga/HELPER": true,
  "name": "takeLatest(boilerplate/UsersPage/GET_USERS, fetchUsers)",
  "next": [Function],
  "return": [Function],
  "throw": [Function],
  Symbol(Symbol.iterator): [Function],
}
`;

exports[`HomePage Saga fetchUsersSagas fetchUsers should match the snapshot for setting token 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://randomuser.me/api/?results=5",
      Object {
        "method": "GET",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`HomePage Saga fetchUsersSagas fetchUsers should set setUsersAction 1`] = `
Object {
  "@@redux-saga/HELPER": true,
  "name": "takeLatest(boilerplate/UsersPage/GET_USERS, fetchUsers)",
  "next": [Function],
  "return": [Function],
  "throw": [Function],
  Symbol(Symbol.iterator): [Function],
}
`;
